extensions/azurePublishNew/src/hooks/useResourceConfiguration.ts

```diff
@@ -110,23 +116,23 @@ export const useResourceConfiguration = () => {
   );
 
   const handleHostNameChange = React.useCallback(
-    (hostName: string) => {
+    (hostName: string, isInvalidName: boolean) => {
       setHostName(hostName);
+      setIsInvalidHostName(isInvalidName);
     },
     [setHostName]
   );
 
-  const stashWizardState = React.useCallback(
-    () =>
-      setItem(getName(), {
-        tenantId,
-        subscriptionId,
-        resourceGroup: { name: resourceGroupName, isNew },
-        deployLocation,
-        luisRegion,
-      }),
-    [tenantId, subscriptionId, resourceGroupName, deployLocation, luisRegion]
-  );
+  const stashWizardState = React.useCallback(() => {
```

Comment:
when your callback has this many dependency, there is no point in using useCallback, it's practically getting redefined as frequent as a regular callback.

Reply:
by not using the useCallback() will trigger an infinite loop, since it is used as dependency for the useEffect() in one of the wizards.

Reply:
When you dont make it useCallback, it wont be a dependency, because it's always up to date.
order:
function definition
effect it's used in